src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/FlowStateAllowedEventService.java

```diff
@@ -493,29 +493,25 @@ public boolean isAllowedOnState(String stateFullName, CaseEvent caseEvent) {
     }
 
     public boolean isAllowedOnStateForSpec(String stateFullName, CaseEvent caseEvent) {
-
         return ALLOWED_EVENTS_ON_FLOW_STATE_SPEC
             .getOrDefault(stateFullName, emptyList())
             .contains(caseEvent);
-
     }
 
     public boolean isAllowed(CaseDetails caseDetails, CaseEvent caseEvent) {
         CaseData caseData = caseDetailsConverter.toCaseData(caseDetails);
+        StateFlow stateFlow = stateFlowEngine.evaluate(caseDetails);
 
         if ((caseData.getSuperClaimType() != null && caseData.getSuperClaimType().equals(SPEC_CLAIM))
             || caseEvent.toString().equals("CREATE_CLAIM_SPEC")) {
-            StateFlow stateFlow = stateFlowEngine.evaluateSpec(caseDetails);
             try {
                 return isAllowedOnStateForSpec(stateFlow.getState().getName(), caseEvent);
             } catch (Exception e) {
                 return false;
```

Comment:
Usually exceptions should be written to the log, at least as debug level so it doesn't pollute production but so we can see if anything weird happens